libs/datamodel/core/src/dml/model.rs

```diff
@@ -210,7 +212,7 @@ impl Model {
         {
             let mut unique_required_fields: Vec<_> = self
                 .scalar_fields()
```

Comment:
I don't see why excluded fields would even show up after calling `.scalar_fields()`. I don't fully get why they land in the data model in the first place, but if they are required to be there, then the wrapper calls on the data model should not return excluded fields in the first place, greatly reducing error surface.